internal/e2e/testdata/time_compare.txt

```diff
@@ -0,0 +1,38 @@
+queryString: time compare\(
+query: (and substr:"time" substr:"compare(")
```

Comment:
neat that this works now :)

Reply:
To check I understand -- will the Sourcegraph query parser convert user input like "func resolve(" into this form, where the parenthesis is escaped?

Reply:
> will the Sourcegraph query parser convert user input like "func resolve(" into this form, where the parenthesis is escaped?

Not exactly. In Sourcegraph, if the pattern `compare(` is labeled as "literal" (which it is if the query is `patternType:keyword time compare(`), [we escape the "(" before parsing it as regexp](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@e3fe897c1d2837a521a4086b30f865c2b6992e65/-/blob/internal/search/zoekt/query.go?L195-200). [Then we convert it to substring query](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@e3fe897c1d2837a521a4086b30f865c2b6992e65/-/blob/internal/search/zoekt/zoekt.go?L31-38). This means, what Zoekt gets is substring query with pattern `"compare("`.

The test here is based on the behavior of Zoekt, which expects valid regex as input.